Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/PrefetchingLib.java

```diff
@@ -146,42 +146,10 @@ public static void init(Context context, int prefetchStrategyNum) {
                     ActivityNode byName = activityGraph.getByName(actName);
                     Long actId = activityMap.get(actName);
 
-                    // Instantiate the static aggregated data for the count of Source --> Destination
-                    // edge visits.  Set up the observers to ensure consistency with the database
-                    if (byName.shouldSetSessionAggregateLiveData()) {
-                        Log.d(LOG_TAG, "loading data for " + actName);
-                        byName.setListSessionAggregateLiveData(
```

Comment:
Invoking the method `setListSessionAggregateLiveData`  (or any of the `setLiveData` method) throws an exception as this is not run in the main thread, which is a requirement to observe the live data object